distro/src/bin/hz

```diff
@@ -20,19 +20,72 @@ PRGDIR=`cd "$PRGDIR"; pwd`
 # set the parent directory as HAZELCAST_HOME.
 export HAZELCAST_HOME=${HAZELCAST_HOME:-$(cd "$PRGDIR/.." >/dev/null; pwd -P)}
 
-if [ $JAVA_HOME ] ; then
-    RUN_JAVA=$JAVA_HOME/bin/java
-else
-    RUN_JAVA=`command -v java 2>/dev/null`
+. "$PRGDIR"/common.sh
```

Comment:
I agree this PR fixes the issue with BETA-1. But I realised one thing - Franta did some changes on the scripts in OS repo (for instance common.sh), which really should require changes in this PR as well. For instance, this is in common.sh in latest master:

```shell
if [ -z "$SCRIPT_DIR" ]; then
  echo "Variable SCRIPT_DIR is expected to be set by the calling script";
  exit 1;
fi;
```

And it's not set here. Currently it works because common.sh in BETA-1 was different. So I suggest either trying to fix this by somehow changing the dependency to SNAPSHOT, or waiting until BETA-2 is released and then finalising the PR.

Reply:
Tried it, fails on start:
```
% ./build/dist/hazelcast-5.0-SNAPSHOT/bin/hz
Variable SCRIPT_DIR is expected to be set by the calling script
```

Reply:
That is a great catch, thank you @mmedenjak, latest commit addresses it, I tested locally with 5.0-SNAPSHOT and it works